ThinDownloadManager/src/main/java/com/thin/downloadmanager/DownloadDispatcher.java

```diff
@@ -216,6 +216,7 @@ private void transferData(HttpURLConnection conn) {
                     errorOccurred = true;
                     updateDownloadFailed(DownloadManager.ERROR_FILE_ERROR,
                         "Error in creating destination file");
+                    return; // escape to finally to skip transferData()
```

Comment:
you are right about the "finally" block called on early return. Generally i tend to avoid early return. I have putup the fix for it. https://github.com/smanikandan14/ThinDownloadManager/commit/1f1ba12b20af999d06167976a1b31f00a4b213a7. Appreciate your changes.